File: frontend/src/types/pathology.ts
/** 병리 분석/보고서 도메인 타입 (병리과). */

/** 지원 염색. */
export type Stain = "HE" | "MT" | "TRI" | "PAS" | "Silver" | "Masson";

export const STAIN_LABEL: Record<Stain, string> = {
  HE:     "H&E",
  MT:     "Masson Trichrome (MT)",
  TRI:    "Masson Trichrome (MT)",
  PAS:    "PAS",
  Silver: "Silver Stain",
  Masson: "Masson Trichrome",
};

/** WSI 위에 토글 가능한 검출 구조 레이어. */
export interface DetectedLayer {
  readonly key: string;
  readonly label: string;
  readonly color: string;
  readonly count: number | null;
  readonly visible: boolean;
}

/** 정량 분석 지표 한 항목. */
export interface QuantMetric {
  readonly key: string;
  readonly label: string;
  readonly value: number | null;
  readonly unit: string;
}

/** 병리 보고서. */
export interface PathologyReport {
  readonly findings: string;
  readonly diagnosis: string;
  readonly status: "draft" | "final";
  readonly updatedAt: string | null;
}

/** 협진 건에 연결된 병리 분석 결과 묶음. */
export interface PathologyResult {
  readonly consultId: string;
  readonly stain: Stain;
  readonly imageUrl: string | null; // Mock — null 이면 업로드/뷰어 안내
  readonly layers: readonly DetectedLayer[];
  readonly metrics: readonly QuantMetric[];
  readonly report: PathologyReport;
}

File: backend/schemas/pathology.py
"""병리 DTO — 프론트 PathologyResult/DetectedLayer/QuantMetric/PathologyReport 대응."""
from schemas.base import CamelModel


class DetectedLayerOut(CamelModel):
    key: str
    label: str
    color: str
    count: int | None = None
    visible: bool


class QuantMetricOut(CamelModel):
    key: str
    label: str
    value: float | None = None
    unit: str


class PathologyReportOut(CamelModel):
    findings: str
    diagnosis: str
    status: str  # draft | final
    updated_at: str | None = None


class PathologyResultOut(CamelModel):
    consult_id: str
    stain: str
    image_url: str | None = None
    layers: list[DetectedLayerOut] = []
    metrics: list[QuantMetricOut] = []
    report: PathologyReportOut


class PathologyReportIn(CamelModel):
    findings: str
    diagnosis: str
    status: str  # draft | final
